test/propagated_params_test.go

```diff
@@ -0,0 +1,382 @@
+				ignoreCondition,
+				ignorePipelineRunStatus,
+				ignoreTaskRunStatus,
+				ignoreConditions,
+				ignoreContainerStates,
+				ignoreStepState,
+			)
+			if d != "" {
+				t.Fatalf(`The resolved spec does not match the expected spec. Here is the diff: %v`, d)
+			}
+			t.Logf("Successfully finished test %q", td.name)
+		})
+	}
+}
+
+func getPropagatedParamPipelineRun(t *testing.T, namespace string) (*v1beta1.PipelineRun, *v1beta1.PipelineRun) {
+	pipelineRun := parse.MustParsePipelineRun(t, fmt.Sprintf(`
+metadata:
+  name: propagated-parameters-fully
+  namespace: %s
+spec:
+  params:
+  - name: HELLO
+    value: "Hello World!"
+  pipelineSpec:
+    tasks:
+      - name: echo-hello
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo $(params.HELLO)
+    finally:
+      - name: echo-hello-finally
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo $(params.HELLO)
+`, namespace))
+	expectedPipelineRun := parse.MustParsePipelineRun(t, fmt.Sprintf(`
+metadata:
+  name: propagated-parameters-fully
+  namespace: %s
+spec:
+  serviceAccountName: default
+  timeout: 1h
+  params:
+  - name: HELLO
+    value: "Hello World!"
+  pipelineSpec:
+    tasks:
+      - name: echo-hello
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo $(params.HELLO)
+    finally:
+      - name: echo-hello-finally
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo $(params.HELLO)
+status:
+  taskRuns:
+    propagated-parameters-fully-echo-hello:
+      pipelineTaskName: echo-hello
+      status:
+        podName: propagated-parameters-fully-echo-hello-pod
+        steps:
+          - name: echo
+            container: step-echo
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo Hello World!
+    propagated-parameters-fully-echo-hello-finally:
+      pipelineTaskName: echo-hello-finally
+      status:
+        podName: propagated-parameters-fully-echo-hello-finally-pod
+        steps:
+          - name: echo
+            container: step-echo
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo Hello World!
+  pipelineSpec:
+    tasks:
+      - name: echo-hello
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo Hello World!
+    finally:
+      - name: echo-hello-finally
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo Hello World!
+`, namespace))
+	return pipelineRun, expectedPipelineRun
+}
+
+func getPropagatedParamWithScopePipelineRun(t *testing.T, namespace string) (*v1beta1.PipelineRun, *v1beta1.PipelineRun) {
```

Comment:
What does Scope mean here?

Reply:
The scope here means that if we have a parameter provided in PipelineRun and pipelineTask, use the value from pipelineTask.